plugins/modules/azure_rm_search.py

```diff
@@ -0,0 +1,480 @@
+        required: true
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+    hosting_mode:
+        description:
+            - Applicable only for the standard3 SKU.
+            - You can set this property to enable up to 3 high density partitions that allow up to 1000 indexes.
+            - For the standard3 SKU, the value is either 'default' or 'highDensity'.
+            - For all other SKUs, this value must be 'default'.
+        choices:
+            - default
+            - highDensity
+        default: 'default'
+    identity:
+        description:
+            - The identity for the resource.
+        choices:
+            - None
+            - SystemAssigned
+        default: 'None'
+    network_rule_set:
+        description:
+            - Network specific rules that determine how the Azure Cognitive Search service may be reached.
+        type: list
+    partition_count:
+        description:
+            - The number of partitions in the search service.
+            - It can be 1, 2, 3, 4, 6, or 12.
+            - Values greater than 1 are only valid for standard SKUs.
+            - For 'standard3' services with hostingMode set to 'highDensity', the allowed values are between 1 and 3.
+        default: 1
+    public_network_access:
+        description:
+            - This value can be set to 'enabled' to avoid breaking changes on existing customer resources and templates.
+            - If set to 'disabled', traffic over public interface is not allowed, and private endpoint connections would be the exclusive access method.
+        choices:
+            - enabled
+            - disabled
+        default: 'enabled'
+    replica_count:
+        description:
+            - The number of replicas in the search service.
+            - It must be a value between 1 and 12 inclusive for standard SKUs.
+            - It must be a value between 1 and 3 inclusive for basic SKU.
+        default: 1
+    sku:
+        description:
+            - The SKU of the Search Service, which determines price tier and capacity limits.
+            - This property is required when creating a new Search Service.
+        choices:
+            - free
+            - basic
+            - standard
+            - standard2
+            - standard3
+            - storage_optimized_l1
+            - storage_optimized_l2
+        default: 'basic'
+    state:
+        description:
+            - Assert the state of the search instance. Set to C(present) to create or update a search instance. Set to C(absent) to remove a search instance.
+        default: present
+        choices:
+            - absent
+            - present
+
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+
+author:
+    - David Duque Hernández (@next-davidduquehernandez)
+'''
+
+EXAMPLES = '''
+  - name: Create Azure Cognitive Search
+    azure_rm_datalakestore:
```

Comment:
Please rewrite this test.
```suggestion
    azure_rm_search:
```